nics-tw/PETsARD
# bug: solve Metadata error in Synthesizer by add col_after_preproc in Metadata when Preprocessor with EncoderOneHot

- See #546 
- 修改方向：
    - Metadata 加一個 `'col_after_expand'`，`to_sdv()` 如果有 `'col_after_expand'` 就改抓這個
    - 我 (Justyn) 想了一下，改叫 `'col_after_preproc'` 更清楚是發生在哪個模組

- 維修結果

![image](https://github.com/nics-tw/PETsARD/assets/8596186/06c2b718-09d5-473c-b223-78d4a720b5dd)

![image](https://github.com/nics-tw/PETsARD/assets/8596186/58f9f08e-c628-4e2a-a671-972375fccecb)

---

- feat
    - PETsARD/processor/base.py
        - append _adjust_metadata as columnwise mode - 0031aedbc1def4b9af9b1e362d22fae62569061f 
        - build new Metadata as col_after_preproc in _adjust_metadata() for MediatorEncoder - 0afd2d2b35d4d5579af1a4081da6a8cf3bc7b4f4 
    - PETsARD/loader/metadata.py
        - use col_after_preproc when it exist in Metadata.to_sdv() - 2253f3c0932fba3b299d2c6846eed5f58f7f5307 

- style
    - PETsARD/processor/encoder.py, mediator.py, missing.py, outlier.py, scaler.py
        - PEP8 - 932a27e3dedf6ca3aaa4a282ca03da26cd8bfbdd 

- docs
    - PETsARD/processor/base.py
        - fix docstring in _align_dtypes() - b3b5421d46520b0265073828aa3e97c86a438f8b 

- test
    - /demo/dev/Issue546.ipynb
        - create Issue546.ipynb - c6808363af177831fdc7fae01e0d4d4c4fdd3703 
        - update testing - 034c0ee757cd8e0ab28f14cf11c29e8cc21d02ac 
        - delete Issue546.ipynb - fcea73a62873b097806a4cb79091c0f5876cd25d 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207341981927488
